### PR TITLE
mate-screensaver: initial commit

### PR DESCRIPTION
**Pull request by cgrzemba**
Add mate-screensaver to oi-userland. So we can replace our outdated xscreensaver in the future.
Thanks to Geoffrey Weiss for contribution

### CODE REVIEW

File: components/desktop/mate/mate-screensaver/Makefile [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+#
+
+MATE_CATEGORY= APP
+BUILD_STYLE= configure
+include ../../../../make-rules/shared-macros.mk
+include $(WS_MAKE_RULES)/mate.mk
+
+COMPONENT_NAME=		mate-screensaver
+COMPONENT_VERSION=	1.26.1
+COMPONENT_MJR_VERSION=	1.26
```
Comment (AndWac) [inline components/desktop/mate/mate-screensaver/Makefile:21]: You only need to set COMPONENT_MJR_VERSION and COMPONENT_MNR_VERSION. COMPONENT_VERSION is set from them in mate.mk.

File: components/desktop/mate/mate-screensaver/Makefile [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+COMPONENT_MNR_VERSION=	1
+COMPONENT_SRC=	$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.xz
+COMPONENT_ARCHIVE_HASH=	\
```
Comment (AndWac) [inline components/desktop/mate/mate-screensaver/Makefile:23]: COMPONENT_SRC and COMPONENT_ARCHIVE are also set in mate.mk.

File: components/desktop/mate/mate-screensaver/Makefile [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	sha256:4fbdb21ea4a59ea8de33ea9bf776d869753e49295604664c30e220e09659b8bc
+COMPONENT_ARCHIVE_URL=	https://pub.mate-desktop.org/releases/$(COMPONENT_MJR_VERSION)/$(COMPONENT_ARCHIVE)
+COMPONENT_PROJECT_URL= https://pub.mate-desktop.org
+COMPONENT_CLASSIFICATION=	Applications/System Utilities
```
Comment (AndWac) [inline components/desktop/mate/mate-screensaver/Makefile:25]: COMPONENT_PROJECT_URL and COMPONENT_ARCHIVE_URL are also set in mate.mk.

File: components/desktop/mate/mate-screensaver/Makefile [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+COMPONENT_LICENSE= GPLv3
+COMPONENT_LICENSE_FILE= COPYING
+COMPONENT_FMRI= desktop/mate/$(COMPONENT_NAME)
+COMPONENT_SUMMARY= Mate Screensaver
```
Comment (AndWac) [inline components/desktop/mate/mate-screensaver/Makefile:28]: COMPONENT_FMRI is also set in mate.mk,
mate.mk has been introduced to set most of the variables automatically.

File: components/desktop/mate/mate-screensaver/Makefile [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+PATH=$(PATH.gnu)
+
+COMPONENT_PREP_ACTION=  cd $(@D) && NOCONFIGURE=1 ./autogen.sh
+
+# FLAVOUR = DBG
+
+CONFIGURE_OPTIONS+=	--sysconfdir=/etc
+CONFIGURE_OPTIONS+=	--libexecdir=$(CONFIGURE_LIBDIR.$(BITS))/mate
+CONFIGURE_OPTIONS+=	--disable-static
```
Comment (AndWac) [inline components/desktop/mate/mate-screensaver/Makefile]: You should check which CONFIGURE_OPTIONS will be set in mate.mk. You can avoid some here.

File: components/desktop/mate/mate-screensaver/Makefile [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+CONFIGURE_OPTIONS+=	--with-console-kit
+CONFIGURE_OPTIONS+=	--enable-locking
+CONFIGURE_OPTIONS+=	--enable-pam
+CONFIGURE_OPTIONS+=	--enable-authentication-scheme=helper
+CONFIGURE_OPTIONS_DBG+=	--enable-debug
+CONFIGURE_OPTIONS+= $(CONFIGURE_OPTIONS_$(FLAVOUR))
+
+CONFIGURE_ENV+= PYTHON="$(PYTHON)"
+
```
Comment (AndWac) [inline components/desktop/mate/mate-screensaver/Makefile]: This is also being set in mate.mk.

File: components/desktop/mate/mate-screensaver/Makefile [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+REQUIRED_PACKAGES += developer/build/autoconf-archive
+REQUIRED_PACKAGES += gnome/theme/background/os-backgrounds
```
Comment (AndWac) [inline components/desktop/mate/mate-screensaver/Makefile:48]: ... and also some REQUIRED_PACKAGES are set in mate.mk

**Comment by AndWac** [comment]
Something seems to be wrong with the dependencies. I got a different sample-manifest here:

 file path=etc/xdg/autostart/mate-screensaver.desktop
 file path=etc/xdg/menus/mate-screensavers.menu
-file path=usr/bin/$(MACH64)/mate-screensaver
-file path=usr/bin/$(MACH64)/mate-screensaver-command
-file path=usr/bin/$(MACH64)/mate-screensaver-preferences
 file path=usr/bin/mate-screensaver
 file path=usr/bin/mate-screensaver-command
 file path=usr/bin/mate-screensaver-preferences
@@ -38,13 +35,6 @@ file path=usr/lib/$(MACH64)/mate/mate-screensaver/mate-screensaver-pam-helper
 file path=usr/lib/$(MACH64)/mate/mate-screensaver/popsquares
 file path=usr/lib/$(MACH64)/mate/mate-screensaver/slideshow
 file path=usr/lib/$(MACH64)/pkgconfig/mate-screensaver.pc
-file path=usr/lib/mate/mate-screensaver-dialog
-file path=usr/lib/mate/mate-screensaver-gl-helper
-file path=usr/lib/mate/mate-screensaver/floaters
-file path=usr/lib/mate/mate-screensaver/mate-screensaver-pam-helper
-file path=usr/lib/mate/mate-screensaver/popsquares
-file path=usr/lib/mate/mate-screensaver/slideshow
-file path=usr/lib/pkgconfig/mate-screensaver.pc
 file path=usr/share/applications/mate-screensaver-preferences.desktop

**Comment by cgrzemba** [comment]
The different manifest is a result of the 64bit only build of mate packages and the relocation the executables in the normal /usr/bin/ path.

**Pull request merged**